pycbc/types/config.py

```diff
@@ -513,17 +525,29 @@ def check_duplicate_options(self, section1, section2, raise_error=False):
                 "Section %s not present in ConfigParser." % (section2,)
             )
 
+        section_and_default = (section1 == f"{section2}-defaultvalues" or
+                               section2 == f"{section1}-defaultvalues")
+
+        if section1.endswith("-defaultvalues") or \
+                section2.endswith("-defaultvalues"):
+            if not section_and_default:
```

Comment:
Why two layers of logic here? Surely if one of the sections has `defaultvalues` it needs to be a subsection of the other, or I don't know what this function is doing.

Reply:
Is it the case of `section1-subsection1` and `section1-defaultvalues` vs `section1` and `section1-defaultvalues`?? (Add comments to explain if so).

Reply:
Maybe the comment here does explain, but I did get confused by this!

Reply:
I have added the comment that it is the section-defaultvalues and section-subsection case that is caught here